app/controllers/sites_controller.rb

```diff
@@ -2,15 +2,23 @@ class SitesController < ApplicationController
   before_action :user_required
 
   def index
-    @sites = Site.order(created_at: :desc).page params[:page]
+    @sites = Site.with_attached_image.includes(:user).order(created_at: :desc).page params[:page]
     @site  = Site.new
   end
 
   def create
     @site = @current_user.sites.build params.fetch(:site, {}).permit(:url)
-    @site.save
+
     respond_to do |format|
-      format.html { redirect_to sites_url }
```

Comment:
Refactored this as I wanted to get errors showing on the form if @site.save fails.

Also left note about potentially doing the same thing for the json response as currently error handling is done within `create.json.jbuilder`